File: spring-mvc/spring-mvc-1/src/main/java/com/springservletthymeleaf/basic/itemservice/domain/item/ItemRequest.java
package com.springservletthymeleaf.basic.itemservice.domain.item;

import lombok.AccessLevel;
import lombok.Builder;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

@Getter
@Setter
@ToString
@EqualsAndHashCode
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class ItemRequest {

	private Long id;

	private String name;

	private int price;

	private int quantity;

	@Builder
	public ItemRequest(String name, int price, int quantity) {
		this.name = name;
		this.price = price;
		this.quantity = quantity;
	}
}

File: spring-mvc/spring-thymeleaf/src/main/kotlin/com/springthymeleaf/message/ItemRequest.kt
package com.springthymeleaf.message

data class ItemRequest(var id: Long?, var name: String?, var price: Int, var quantity: Int) {
  constructor() : this(null, null, 0, 0)
}

/*
GenericMapper<D, E> 사용 불가
kapt 문제인 듯 싶음, 코틀린 바보 자식
직접 타입 박아줘서 만들어야 하는디 실제로 쓰기에는 중복이 너무 많을듯
 */